### ISSUE TITLE
Buffer swithching commands with ordinal numbers

### ISSUE THREAD

**Issue by cenk** [ap/vim-buftabline#17]
I am used to switch buffers by specifying their numbers instead of pressing ```:bn``` or ```:bp``` shortcut key repeatedly. There is an implementation of this feature in airline which I am currently using:

https://github.com/vim-airline/vim-airline/blob/30f078daf569e7d5e4f7829e39316387af349b41/doc/airline.txt#L555

Is it possible to add this feature in vim-buftabline?

Would you merge it if I make a PR?

**Comment by ap** [comment]
I’m not clear on what your feature request actually is.

Are asking for `let g:buftabline_numbers = 2`? Because that’s already implemented.

Or do you mean basically adding the VimL code from #11 to the plugin directly, adding key bindings to it? In which case, I am not likely to take a PR, but that doesn’t stop you from solving your own need by just putting that in your `.vimrc`.

The reason I’m not enthused is I don’t think buftabline can do a good enough job of it, for two reasons:

1. I am hesitant to pick default keybindings for everyone, because they will collide with some users’ own bindings. So they must come with a preference to turn off butabline’s bindings. I don’t like adding on/off preferences.

2. If I do pick defaults, they must be picked so well that most people can just use them without thinking about them. I simply don’t know the UI conventions across systems well enough. I’m on a Mac myself and the (largely consistent) native convention there is to put this function on ⌘1 ⌘2 ⌘3 etc – which cannot work for Windows or Linux people, as they don’t have that key, and the extra “Windows” key they do (usually) have is not used at all like the Mac’s Command key. It’s unclear to me whether they even have a similarly universal convention.

*However*, some fraction of users evidently want/use such keybindings, so it will be useful to support that somehow.

* A no-brainer would be documentation about how to add that in one’s `vimrc`, with code examples, so users don’t need to figure it out by source-diving.

* Some syntactic relief to make it less ugly might possibly be worthwhile, too.

  Actually, come to think of it – yes, there is definitely something that buftabline can provide, within the limits given above: [`<Plug>` mappings](http://vimdoc.sourceforge.net/htmldoc/usr_41.html#using-<Plug>) for switching buffers.

I’d take patches for these. (Though you may have to tolerate me amending them because I’m fussy that way… in case you hadn’t caught on to that. :smile:)

**Comment by ap** [comment]
Which, d’oh, is actually right in there in the code you linked to. I was just so blinded by the issue of Airline having default keybindings that I completely missed that it does that by way of `<Plug>` mappings.

So actually, the answer is – yes, mostly.

**Comment by ap** [comment]
And another d’oh as I find it doesn’t even do that. So that means the answer is just yes. Sorry I had to refactor your request in order to understand it.

**Comment by ap** [comment]
So let’s start over and try this again:

There’s code in the thread of #11 for that, but I didn’t think of `<Plug>` mappings for it. That would be a useful feature to ship. I won’t ship default keybindings with the plugin because I don’t think it can do a good enough job of picking defaults, and it would require another on/off preference, which I don’t like adding.

So, I’ll take patches for that – though you may have to tolerate me amending the patches. :smille: But I’ll ship the feature regardless of whether you provide a patch for it.

**Comment by MiguelLatorre** [comment]
+1
 I was about to ask for this too.

**Comment by cenk** [comment]
Fixed in a3e32bd322b909f20f87a2c7ddabf6416dc21cec

**Comment by ap** [comment]
Ah yeah, sorry, forgot to close this.

And sorry it took me so long to get around to pushing it out. The plug names derailed me.

The reason I ended up changing them is because I wanted to have a plug for the 10th buffer. But adding `<Plug>BufTabLineSelectBuffer10` mysteriously broke `<Plug>BufTabLineSelectBuffer1`. Actually of course, it didn’t, it just made it ambiguous, so Vim would wait to see if the next character was a 0 before executing the mapping. It took me a day or so to figure that out… embarrassingly long anyway.

The solution is obvious: putting a non-numeral after the buffer number removes the ambiguity and then Vim can execute the mapping immediately. But that meant I had to pick some other name, and then I got sidetracked thinking about which choice I liked best…

And then I got busy with other stuff and forgot about this patch for a while.

Anyway, hope the wait was worth it. Share and enjoy. :smile:
